src/plugins/yarn.js

```diff
@@ -61,11 +34,9 @@ export default class Yarn {
     if (!this.bin) throw new Error('yarn not found')
     const execa = require('execa')
     this.out.debug(`${options.cwd}: ${this.bin} ${args.join(' ')}`)
-    deleteYarnRoadrunnerCache()
     let unlock = await lock.write(this.lockfile)
-    let output = await execa(this.bin, args, options)
+    let output = await execa('node', [this.bin, ...args], options)
```

Comment:
@dickeyxxx could you have this [fork](https://nodejs.org/api/child_process.html#child_process_child_process_fork_modulepath_args_options) instead so that it reuses the interpreter rather than expecting node to be in the path?  Another valid option would be to use the full path of the heroku node interpreter here if available.